test/unit/datasets/test_datasets.py

```diff
@@ -52,8 +54,19 @@ def test_dataset_as_bunch(self, as_frame, fetch_function):
         ],
     )
     def test_dataset_as_X_y(self, as_frame, fetch_function):
+        if as_frame is False and fetch_function is fetch_diabetes_hospital:
```

Comment:
nit: `pytest.skip` could be used here I think because `return` marks it as passed.

Reply:
Fixed, thanks for the nit 😌